### PR TITLE
ASTF program adds rx_mode in UDP keepalive.

### PR DESCRIPTION
**Pull request by jsmoon**
Hi, this PR is to add RX-only mode to the ASTF UDP keepalive feature.

Since the keepalive timer has been reset by default with TX/RX packets, during a flow sends packets by `send_msg`, the keepalive timeout would not occur even though there are no RX packets.
In this situation, I'd like to check the keepalive timeout by RX packets only. So I added the `rx_mode` to the `set_keepalive_msg`.

@hhaim, please review my changes and give your feedback.

### CODE REVIEW

**Comment by jsmoon** [comment]
@hhaim could you look into the trex-bot failure?
```
  File "/usr/lib64/python2.7/httplib.py", line 881, in _send_output
    self.send(msg)
  File "/usr/lib64/python2.7/httplib.py", line 843, in send
    self.connect()
  File "/usr/lib64/python2.7/httplib.py", line 824, in connect
    self.timeout, self.source_address)
  File "/usr/lib64/python2.7/socket.py", line 571, in create_connection
    raise err
socket.error: [Errno 111] Connection refused
```

**Comment by hhaim** [comment]
> @hhaim could you look into the trex-bot failure?
> 
> ```
>   File "/usr/lib64/python2.7/httplib.py", line 881, in _send_output
>     self.send(msg)
>   File "/usr/lib64/python2.7/httplib.py", line 843, in send
>     self.connect()
>   File "/usr/lib64/python2.7/httplib.py", line 824, in connect
>     self.timeout, self.source_address)
>   File "/usr/lib64/python2.7/socket.py", line 571, in create_connection
>     raise err
> socket.error: [Errno 111] Connection refused
> ```

will be fixed, there are some sort of PR that can crash the daemon. 

For the PR, I don't understand the need for this, 
if the template is a tx only, why would you need a keep alive on the rx side? 
If the template is a rx/tx the keep alive will work implicitly (from the last tx packet to the next rx packet).
Why would you want to disable keep alive in a duration of tx? 
so to you question, the timeout **will** occur but in a delay

**Comment by hhaim** [comment]
recheck

**Comment by jsmoon** [comment]
@hhaim, I have a profile that simulates VoIP traffic behavior.
It sends UDP packets in both directions. (client -> server and server -> client)
The client and server only send packets at a given rate without receiving.
Even though one direction is dropping packets over the limitation(=keepalive time),
there is no flow close by the keepalive timeout due to the sending packets.
I'd like to make the flow closed by the keepalive timeout.

**Comment by hhaim** [comment]
recheck

**Comment by hhaim** [comment]
@jsmoon understood.

**Comment by hhaim** [comment]
@jsmoon I would progress the astf.minor version.

File: src/astf/astf_db.cpp [hunk 1]
```diff
@@ -574,6 +574,9 @@ void CAstfDB::fill_keepalive_pkt(uint32_t program_index,
     assert_cmd(program_index, cmd_index, "keepalive", cmd);
 
     res.u.m_keepalive.m_keepalive_msec =cmd["msec"].asUInt();
+    if (cmd["rx_mode"] != Json::nullValue) {
+        res.u.m_keepalive.m_rx_mode = cmd["rx_mode"].asBool();
+    }
```
[the remark this answers is not in the document]
Comment (jsmoon) [reply src/astf/astf_db.cpp:579]: I missed the point. Thank you!

**Comment by hhaim** [comment]
rebased

**Pull request closed without merging**